state.go

```diff
@@ -272,6 +272,12 @@ func (m *Memberlist) probeNode(node *nodeState) {
 	// soon as possible.
 	deadline := sent.Add(probeInterval)
 	addr := node.Address()
+
+	// Arrange for our self-awareness to get updated.
+	var awarenessDelta int
+	defer func() {
+		m.awareness.ApplyDelta(awarenessDelta)
+	}()
```

Comment:
This is moved up from below, because we might've jump over the place where the call was previously. At this time `awarenessDelta` is `0` and nothing happens if it is applied.